---
title: "Template Switching in the litterature"
subtitle: "RT optimisation with the Labcyte Echo 525: bibliography"
author: "Charles Plessy"
date: "18 May 2018"
output:
  html_document:
    keep_md: yes
    toc: yes
    pandoc_args:
     - "--lua-filter=links-to-markdown.lua"
---


Summary table
=============

|TSO (nM)|RTP (nM)|dNTP (mM)|Mg (mM)|  Mn | DTT | Enzyme      | Reference          |
|-------:|-------:|--------:|------:|----:|----:|------------:|-------------------:|
|  10,000|   1000 |     2.5 |    3.0|   0 |  10 |  200 (SSIII)|   [nanoCAGE 2017][]|
|   1,000|   1000 |     2.0 |    7.2|   0 |   4 |   50 (SSII) |        [STA 2017][]|
|   4,600|    135 |     1.2 |    8.7|   0 |  5.4|   6.8 (SSII)|[STRT-seq-2i 2017][]|
|   1,000|100 or 1|     1.0 |     1×|   0 |  2.4|  SmartScribe|       [CATS 2014][]|
|   1,000|   1000 |     1.0 |   12.0|   0 |  2.5|  100 (SSII) |      [Smart-seq2][]|
|   1,000|GSP 250 |     0.5 |    6.0|   3 |  10 | RevertAid H |       [RACE 2010][]|
|     200|    200 |     1.0 |    3.0|   3 |   1 |   5 (SSII)  |       [STRT 2011][]|
|       0|    750 |     4.0 |    3.0|   2 |   5 | 200 (SSII)  |  [CapSelect 1999][]|
|       0|  65789 |     1.1 |    3.0|   0 |   5 | 760 (SSIII) |  nAnTiCAGE|


Smart-seq2
==========

Assuming a final volume of 10 μl:

[Smart-seq2][]: Picelli et al.,  Nat Methods. 2013 Nov;10(11):1096-8. doi:10.1038/nmeth.2639

Total RNA or single-cell lysates mixed with
   1 μl of anchored oligo-dT primer (10 μM)
   1 μl of dNTP mix (10 mM, Fermentas),
denatured at 72 °C for 3 min and immediately placed on ice afterwards.
7 μl of the first-strand reaction mix, containing
   0.50 μl SuperScript II reverse transcriptase (200 U/μl, Invitrogen),
   0.25 μl RNAse inhibitor (40 U/μl, Clontech),
      2 μl Superscript II First-Strand Buffer (5×, Invitrogen),
   0.25 μl DTT (100 mM, Invitrogen),
      2 μl betaine (5 M, Sigma),
    0.9 μl MgCl2 (100 mM, Sigma),
      1 μl TSO (10 μM) and
    0.1 μl nuclease-free water (Gibco), were added to each sample.

Reverse transcription reaction was carried out by incubating at 42 °C for 90 min, followed by 10 cycles of (50 °C for 2 min, 42 °C for 2 min). Finally, the reverse transcriptase was inactivated by incubation at 70 °C for 15 min.


RACE 2010
=========

See table in paper:

[RACE 2010][]: 


STRT-seq-2i
===========

[STRT-seq-2i 2017][]: Hochgerner et al, Sci Rep. 2017 Nov 27;7(1):16327. doi:10.1038/s41598-017-16546-4

50 nL cell suspension plus
50 nL lysis mix (500 nM RT primer,
                 4.5 mM dNTP,
                  2% Triton-X-100,
                  20 mM DTT,
                 1.5 U/μl TaKaRa RNase Inhibitor),
plus 85 nL RT mix (2.1× SuperScript II First-Strand Buffer,
                   12.6 mM MgCl2,
                   1.79 M betaine,
                   14.7 U/μl SuperScript II,
                   1.58 U/μl TaKaRa RNase Inhibitor,
                   10.5 μM P1B-UMI-RNA-TSO)


CATS
====

[CATS 2014][]: Turchinovich et al, RNA Biol. 2014;11(7):817-28. doi:10.4161/rna.29304

For the reverse transcription,
    1 μl of poly(A) tailed RNA or poly(dA) tailed DNA was mixed with
  2.5 μl of 1x First-Strand RT buffer containing 20% DMSO and
    1 μl of poly(dT) primer (final concentration 0.1 μM for 1 ng and 0.001 μM for 5 pg of RNA or DNA). The entire solution was incubated at 72 °C for 2 min and then cooled to 42 °C for 1 min.
In the following step a master mix containing
    2 μl 5x First-Strand RT buffer (Clontech),
    1 μl dNTP (10 mM each),
    1 μl SmartScribe RT polymerase (Clontech),
 0.25 μl DTT (100 mM) and
 0.25 μl of Recombinant RNase Inhibitor (Clontech) was added to the DNA(RNA)/primer solution and incubated for 15 min at 42 °C.
Next,
    1 μl of 10 μM 5′-biotin blocked template switch oligonucleotide (TSO) GTT CAG AGT TCT ACA GTC CGA CGA TC rGrGrG was added to the RT reaction and incubated for another 15 min at 42 °C.
    

CapSelect
=========

[CapSelect 1999][] Schmidt et al, Nucleic Acids Res. 1999 Nov 1;27(21):e31.

reaction mixture (20 μl)
containing 50 mM Tris–HCl pH 8.3,
           75 mM KCl,
            3 mM MgCl2,
            5 mM DTT,
       1 mM each dNTPs,
       0.1 mg/ml BSA,
            20 U RNase inhibitor
           200 U SuperScriptII,
        0.75 µM RTP.
Addition of 0.4 μl fresh MnCl2 solution (100 mM)


nanoCAGE 2017
=============

[nanoCAGE 2017][] Poulain et al, Methods Mol Biol. 2017;1543:57-109. doi:10.1007/978-1-4939-6716-2_4

Final RT reaction: 1× SSIII buffer (3 mM MgCl2,
                                    75 mM KCl,
                                    50 mM Tris-HCl pH 8.3),
                10 mM DTT,
               2.5 mM dNTP,
               0.75 M betaine,
                200 U SuperScript III,
                 1 µM RTP
                10 µM TSO.

STRT
====

[STRT 2011][], Islam et al., Genome Res. 2011 Jul;21(7):1160-7. doi:10.1101/gr.110882.110

STRT buffer: 20 mM Tris-HCl pH 8.0,
             75 mM KCl,
              6 mM MgCl2,
             0.02% Tween-20,
            400 nM RT primer,
            400 nM TSO.

RT buffer: 4 mM DTT,
           2 mM dNTP,
         5 U/μL Superscript II,
         100 mM Tris-HCl pH 8,
         375 mM KCl,
           0.1% Tween-20,
           6 mM MnCl2
         
Final reaction in STRT/RT buffer 1/1.


STA
===

[STA 2017][]: Lee et al, BMC Biol. 2017 Mar 21;15(1):22. doi:10.1186/s12915-017-0359-5

4 μl of 1× RT buffer (50 mM Tris-HCl pH 8.3,
                      75 mM KCl,
                       3 mM MgCl2,
                      10 mM dithiothreitol (DTT))
    100 μM ATP,
       1 U SUPERase In,
    1.25 U PAP
    
add     0.5 μl of 20 μM RTP
          2 μl 5 M betaine,
        0.5 μl 20 μM TSO,
        0.6 μl 100 mM MgCl2,
        0.8 μl 25 mM dNTP mixture,
        0.6 μl 10× RT buffer,
        0.4 μl 100 mM DTT,
        0.3 μl 0.1% Triton X-100,
       0.25 μl 200 U/μl superscript II,
       0.05 μl 20U/μl SUPERase In.

[CapSelect 1999]: https://pubmed.gov/10518626 "CapSelect: a highly sensitive method for 5' CAP-dependent enrichment of full-length cDNA in PCR-mediated analysis of mRNAs."
[STRT 2011]: https://pubmed.gov/21543516 "Characterization of the single-cell transcriptional landscape by highly multiplex RNA-seq."
[Smart-seq2]: https://pubmed.gov/24056875 "Smart-seq2 for sensitive full-length transcriptome profiling in single cells."
[CATS 2014]: https://pubmed.gov/24922482 "Capture and Amplification by Tailing and Switching (CATS). An ultrasensitive ligation-independent method for generation of DNA libraries for deep sequencing from picogram amounts of DNA and RNA."
[STA 2017]: https://pubmed.gov/28327113 "Low-cell-number, single-tube amplification (STA) of total RNA revealed transcriptome changes from pluripotency to endothelium."
[nanoCAGE 2017]: https://pubmed.gov/28349422 "NanoCAGE: A Method for the Analysis of Coding and Noncoding 5'-Capped Transcriptomes."
[STRT-seq-2i 2017]: https://pubmed.gov/29180631 "dual-index 5' single cell and nucleus RNA-seq on an addressable microwell array."