---
title: "Labcyte-RT Data QC (3rd experiment, plates 1, 2 and 3)"
output:
  html_document:
    keep_md: yes
    number_sections: yes
    toc: yes
---

```{r echo = FALSE}
knitr::opts_chunk$set(cache  = TRUE)
knitr::opts_knit$set(verbose = TRUE)
ggplot2::theme_set(ggplot2::theme_bw())
```


Experiment 3, plates 1, 2 and 3
===============================

The reverse-transcriptase was SuperScript III.

Plates 1~3 and 4~6 were processed separately because the Moirai pipeline does
not handle dual indexing.

For merging multiple experiments, the following unique plate IDs `E`, `F` and
`G` are assigned to the replicates `1`, `2`, and `3` respectively.

This QC report is slim on purpose; more plots will be produced in the combined
analysis.


Load R packages
===============

```{r load_R_libraries, message = FALSE}
library("CAGEr")
library("ggplot2")
library("magrittr")
library("MultiAssayExperiment")
library("SummarizedExperiment")
```


Load CAGE Data
===================

Load summary statistics from MOIRAI and polish the names
--------------------------------------------------------

### Define MOIRAI shortcuts

```{r moirai_shortcuts, dependson = "load_R_libraries"}
MISEQ_RUN      <- "180403_M00528_0348_000000000-B4GP8_p123"
WORKFLOW       <- "OP-WORKFLOW-CAGEscan-short-reads-v2.1~rc1"
MOIRAI_STAMP   <- "20180409105009"
MOIRAI_PROJ    <- "project/Labcyte" 
MOIRAI_USER    <- "nanoCAGE2" 
ASSEMBLY       <- "mm9"
BASEDIR        <- "/osc-fs_home/scratch/moirai"
MOIRAI_BASE    <- file.path(BASEDIR, MOIRAI_USER)
MOIRAI_RESULTS <- file.path(MOIRAI_BASE, MOIRAI_PROJ, paste(MISEQ_RUN, WORKFLOW, MOIRAI_STAMP, sep = "."))
```

### Load MOIRAI statistics

```{r MOIRAI, dependson = "moirai_shortcuts"}
ce <- smallCAGEqc::loadMoiraiStats(
  pipeline  = WORKFLOW,
  multiplex = file.path( MOIRAI_BASE, "input", paste0(MISEQ_RUN, ".multiplex.txt")),
  summary   = file.path( MOIRAI_RESULTS, "text", "summary.txt")) %>% DataFrame

ce$inputFiles <- paste0(MOIRAI_RESULTS, "/CAGEscan_fragments/", ce$samplename, ".bed")

# Discard lines for which input files do not exist.
ce <- ce[sapply(ce$inputFiles, file.exists),]

# Discard lines for which input files are empty.
ce <- ce[file.info(ce$inputFiles)$size != 0,]

ce$inputFilesType <- c("bed")
ce$sampleLabels <- as.character(ce$samplename)

# Replace indexes in group names by RNA amounts extracted from sample sheet.
levels(ce$group) <- system("grep TATCCTCT 180403_M00528_0348_000000000-B4GP8.SampleSheet.csv | cut -f 6,10 -d , | grep g_ | sort | cut -f2 -d, | cut -f1 -d_", intern = TRUE)

# Sort the levels by RNA amount
ce$group %<>% factor(levels = c("100ng", "10ng", "1ng", "100pg", "10pg"))

ce$repl <- ce$index
levels(ce$repl) <- system("grep TATCCTCT 180403_M00528_0348_000000000-B4GP8.SampleSheet.csv | cut -f 6,10 -d , | grep g_ | sort | cut -f 2 -d _ | sed 's/\r//'", intern = TRUE)
ce$repl %<>% factor(levels = 1:3)

# Define plate IDs
ce$plateID <- ce$repl
levels(ce$plateID) <- c("E", "F", "G")
rownames(ce) %<>% paste(ce$plateID, sep = "_")
ce$sampleLabels <- rownames(ce)

ce
```


Load plate design
-----------------

Using plate 2 design, see [Labcyte-RT2.md](Labcyte-RT2.md).

```{r PLATE, dependson = "MOIRAI"}
plate <- read.table("plate2.txt", sep = "\t", header = TRUE)
ce %<>% cbind(plate[match( paste(ce$barcode, ce$group)
                         , paste(plate$BARCODE_SEQ, plate$RNA_group)), ])
rm(plate)
```


Create a CAGEexp object and load expression data
------------------------------------------------

```{r new_CAGEexp, message = F, echo = FALSE, dependson = "PLATE"}
ce <- CAGEexp( colData    = ce
             , genomeName = "BSgenome.Mmusculus.UCSC.mm9")
```

```{r get_CTSS_data, dependson = "new_CAGEexp"}
getCTSS(ce, useMulticore = TRUE)
removeStrandInvaders(ce)
```


Save the CAGEexp file
---------------------

```{r save_myCAGEexp_object, dependson = "get_CTSS_data"}
saveRDS(ce, "Labcyte-RT_Data_Analysis_3a.Rds")
```


Quality controls
================

Custom _scopes_ displaying _strand invasion_ artefacts.

```{r custom_plotAnnot_scopes}
msScope_qcSI <- function(libs) {
  libs$Tag_dust     <- libs$extracted   - libs$rdna - libs$spikes - libs$cleaned
  libs$rDNA         <- libs$rdna
  libs$Spikes       <- libs$spikes
  libs$Unmapped     <- libs$cleaned     - libs$mapped
  libs$Non_proper   <- libs$mapped      - libs$properpairs
  libs$Duplicates   <- libs$properpairs - libs$librarySizes - libs$strandInvaders
  libs$Invaders     <- libs$strandInvaders
  libs$Counts       <- libs$librarySizes
  list( libs    = libs
      , columns = c( "Tag_dust", "rDNA", "Spikes", "Unmapped"
                   , "Non_proper", "Duplicates", "Invaders", "Counts")
      , total   = libs$extracted)
}

msScope_counts <- function(libs) {
  libs$Promoter   <- libs$promoter
  libs$Exon       <- libs$exon
  libs$Intron     <- libs$intron
  libs$Intergenic <- libs$librarySizes - libs$promoter - libs$intron - libs$exon
  libs$Invaders   <- libs$strandInvaders
  list( libs    = libs
      , columns = c("Promoter", "Exon", "Intron", "Intergenic", "Invaders")
      , total   = libs$librarySizes + libs$strandInvaders)
}
```

By RNA input
------------

Negative controls with no RNA gave much less sequences than the regular
samples with RNA.

```{r qc_RNA_quantity, dependson = "get_CTSS_data"}
ggpubr::ggarrange( legend = "right", common.legend = TRUE,
  plotAnnot( ce, scope = msScope_qcSI, group = "plateID", normalise = FALSE
           , title = "sequence counts"
           , facet="RNA") +
    facet_wrap(~facet, ncol = 1),
  plotAnnot( ce, scope = msScope_qcSI, group = "plateID", normalise = TRUE
         , title = "normalised to 100 %"
         , facet="RNA") +
  facet_wrap(~facet, ncol = 1)
) %>% ggpubr::annotate_figure(top="QC of processing, by ng of input RNA")
```

The normalised QC profile is not much different, therefore
the sequences might be _in silico_ or _in vitro_ contaminations.


Annotation with GENCODE
-----------------------

Collect Gencode annotations and gene symbols via a local GENCODE file
(mm9 gencode not available in AnnotationHub)

Most pairs align in intergenic regions.  Is it because of the sequence error
combined with very short read 1s ?

```{r annotate_CTSS, dependson = "get_CTSS_data"}
annotateCTSS(ce, rtracklayer::import.gff("/osc-fs_home/scratch/gmtu/annotation/mus_musculus/gencode-M1/gencode.vM1.annotation.gtf.gz"))
```

```{r CTSS_annotation_plot_per_replicate, dependson = "annotate_CTSS"}
ggpubr::ggarrange( legend = "right", common.legend = TRUE,
  plotAnnot( ce, scope = "counts", group = "plateID"
           , title = "Sequence counts"
           , facet = "RNA", normalise = FALSE) +
    facet_wrap("facet", ncol = 1),
  plotAnnot( ce, scope = "counts", group = "plateID"
           , title = "Normalised to 100%"
           , facet = "RNA", normalise = TRUE) +
    facet_wrap("facet", ncol = 1)
)  %>% ggpubr::annotate_figure(top="QC annotation, by ng of input RNA")
```


By TSO concentration
--------------------

Problem with TSO == 60 ??  Mabye the stock primer had a defect.

```{r TSO_qc_raw, dependson = "get_CTSS_data"}
plotAnnot(ce, scope = msScope_qcSI, group = "TSO", normalise = FALSE, facet = "RNA", title = "QC, sequence counts, by RNA quantity")
```

```{r TSO_qc_norm, dependson = "get_CTSS_data"}
plotAnnot(ce, scope = msScope_qcSI, group = "TSO", normalise = TRUE, facet = "RNA", title = "QC, normalised to 100 %, by RNA quantity")
```

```{r TSO_annot_raw, dependson = "get_CTSS_data"}
plotAnnot(ce, scope = msScope_counts, group = "TSO", normalise = FALSE, facet = "RNA", title = "QC annotation, normalised to 100 %, by RNA quantity")
```

```{r TSO_annot_norm, dependson = "get_CTSS_data"}
plotAnnot(ce, scope = msScope_counts, group = "TSO", normalise = TRUE, facet = "RNA", title = "QC annotation, normalised to 100 %, by RNA quantity")
```



Session information
===================

```{r}
sessionInfo()
```