---
title: "Labcyte-RT Data Analysis (2nd experiment)"
output:
  html_document:
    keep_md: yes
    number_sections: yes
    toc: yes
---

```{r echo = FALSE}
knitr::opts_chunk$set(cache  = TRUE)
knitr::opts_knit$set(verbose = TRUE)
ggplot2::theme_set(ggplot2::theme_bw())
```

Data load and QC in R
=====================

```{r load_R_libraries, message = FALSE}
library("CAGEr")
library("ggplot2")
library("magrittr")
library("MultiAssayExperiment")
library("RColorBrewer")
library("SummarizedExperiment")
library("vegan")
```

MOIRAI shortcuts

```{r moirai_shortcuts}
MISEQ_RUN      <- "180123_M00528_0325_000000000-B4PCK"
WORKFLOW       <- "OP-WORKFLOW-CAGEscan-short-reads-v2.1~rc1"
MOIRAI_STAMP   <- "20180124102551"
MOIRAI_PROJ    <- "project/Labcyte" 
MOIRAI_USER    <- "nanoCAGE2" 
ASSEMBLY       <- "mm9"
BASEDIR        <- "/osc-fs_home/scratch/moirai"
MOIRAI_BASE    <- file.path(BASEDIR, MOIRAI_USER)
MOIRAI_RESULTS <- file.path(MOIRAI_BASE, MOIRAI_PROJ, paste(MISEQ_RUN, WORKFLOW, MOIRAI_STAMP, sep = "."))
```

Load CAGE libraries
===================

Load summary statistics from MOIRAI and polish the names
--------------------------------------------------------

```{r MOIRAI}
ce <- smallCAGEqc::loadMoiraiStats(
  pipeline  = WORKFLOW,
  multiplex = file.path( MOIRAI_BASE, "input", paste0(MISEQ_RUN, ".multiplex.txt")),
  summary   = file.path( MOIRAI_RESULTS, "text", "summary.txt")) %>% DataFrame

ce$inputFiles <- paste0(MOIRAI_RESULTS, "/CAGEscan_fragments/", ce$samplename, ".bed")

# Discard lines for which input files do not exist.
ce <- ce[sapply(ce$inputFiles, file.exists),]

# Discard lines for which input files are empty.
ce <- ce[file.info(ce$inputFiles)$size != 0,]

ce$inputFilesType <- c("bed")
ce$sampleLabels <- as.character(ce$samplename)
ce

# Replace indexes in group names by RNA amounts extracted from sample sheet.
levels(ce$group) <- system("cut -f 6,8 -d , 180123_M00528_0325_000000000-B4PCK.SampleSheet.csv | grep g_ | sort | cut -f2 -d, | cut -f1 -d_", intern = TRUE)

# Sort the levels by RNA amount
ce$group %<>% factor(levels = c("100ng", "10ng", "1ng", "100pg", "10pg"))

ce$repl <- ce$index
levels(ce$repl) <- system("cut -f 6,8 -d , 180123_M00528_0325_000000000-B4PCK.SampleSheet.csv | grep g_ | sort | cut -f 2 -d _ | sed 's/\r//'", intern = TRUE)

ce$group_repl <- paste(ce$group, ce$repl) %>%
                  factor(levels = c( paste("100ng", 1:3),   paste("10ng", 1:3)
                                   , paste("1ng",   1:3),   paste("100pg", 1:3)
                                   , paste("10pg",  1:3)))
```

```{r PLATE}
plate <- read.table("plate.txt", sep = "\t", header = TRUE)
plate$TSO %<>% factor
ce %<>% cbind(plate[match( paste(ce$barcode, ce$group)
                         , paste(plate$BARCODE_SEQ, plate$RNA_group)), ])
```


Create a CAGEexp object and load expression data
------------------------------------------------

```{r new_CAGEexp, message = F, echo = FALSE}
ce <- CAGEexp( colData    = ce
             , genomeName = "BSgenome.Mmusculus.UCSC.mm9")
```

```{r get_CTSS_data}
getCTSS(ce, useMulticore = TRUE)
removeStrandInvaders(ce)
```

Quality controls
----------------

Custom _scopes_ displaying _strand invasion_ artefacts.

```{r custom_plotAnnot_scopes}
msScope_qcSI <- function(libs) {
  libs$Tag_dust     <- libs$extracted   - libs$rdna - libs$spikes - libs$cleaned
  libs$rDNA         <- libs$rdna
  libs$Spikes       <- libs$spikes
  libs$Unmapped     <- libs$cleaned     - libs$mapped
  libs$Non_proper   <- libs$mapped      - libs$properpairs
  libs$Duplicates   <- libs$properpairs - libs$librarySizes - libs$strandInvaders
  libs$Invaders     <- libs$strandInvaders
  libs$Counts       <- libs$librarySizes
  list( libs    = libs
      , columns = c( "Tag_dust", "rDNA", "Spikes", "Unmapped"
                   , "Non_proper", "Duplicates", "Invaders", "Counts")
      , total   = libs$extracted)
}

msScope_counts <- function(libs) {
  libs$Promoter   <- libs$promoter
  libs$Exon       <- libs$exon
  libs$Intron     <- libs$intron
  libs$Intergenic <- libs$librarySizes - libs$promoter - libs$intron - libs$exon
  libs$Invaders   <- libs$strandInvaders
  list( libs    = libs
      , columns = c("Promoter", "Exon", "Intron", "Intergenic", "Invaders")
      , total   = libs$librarySizes + libs$strandInvaders)
}

```

### By RNA input

Negative controls with no RNA gave much less sequences than the regular
samples with RNA.

```{r qc_RNA_quantity_raw}
plotAnnot( ce, scope = msScope_qcSI, group = "RNA", normalise = FALSE
         , title = "QC control, by ng of input RNA (sequence counts)")
```

The normalised QC profile is not much different, therefore
the sequences might be _in silico_ or _in vitro_ contaminations.

```{r qc_RNA_quantity_norm}
plotAnnot( ce, scope = msScope_qcSI, group = "RNA", normalise = TRUE
         , title = "QC control, by ng of input RNA (normalised to 100 %)")
```

### Removal of the RNA negative controls

To ease data handling, the negative controls with no RNA are removed.

```{r}
ce <- ce[,ce$RNA != 0]
```

### By RT primer quantity

Strangely, libraries made with no RT primers have a QC profile that is not
dramatically different from other libraries.  This might again be explained
by contaminations, although the amount of sequences in the "no RT primer"
samples is a bit high for such an explanation.

```{r qc_rt_primer_quantity_raw}
plotAnnot( ce, scope = msScope_qcSI, group = "RT_PRIMERS", normalise = FALSE
         , title = "QC control, by amount of RT primers (in Î¼M)")
```

```{r qc_rt_primer_quantity_norm}
plotAnnot( ce, scope = msScope_qcSI, group = "RT_PRIMERS", normalise = TRUE
         , title = "QC control, by amount of RT primers (normalised to 100 %)")
```

### Removal of the primer-negative controls

To ease data handling (for instance when working with primer ratios), the
negative controls with no primers are removed.

```{r}
ce <- ce[,ce$RT_PRIMERS != 0]
ce$PRIMERS_RATIO %<>% droplevels
ce$PRIMERS_RATIO %<>%
  factor(levels = c( "0.078125", "0.15625", "0.3125", "0.625", "1.25", "2.5"
                   , "5", "10", "20", "40", "80", "160", "320", "640"))
```

### By group and replicates

In all conditions, there are a large number of unmapped sequences.  What are they ?

```{r}
plotAnnot(ce, scope = msScope_qcSI, group = "repl", facet="group") +
  facet_wrap(~facet, nrow = 5)
plotAnnot(ce, scope = msScope_qcSI, group = "repl", normalise = FALSE, facet="group") +
  facet_wrap(~facet, nrow = 5)
```

There is a trend decreasing sequence yield with decreasing TSO amounts.
Interestingly, the optimum seems to be lower when RNA quantity is lower.

But what is wrong with TSO == 40 ??  (and to some extent with TSO == 1.25)

```{r}
plotAnnot(ce, scope = msScope_qcSI, group = "TSO", normalise = FALSE, facet = "group")
plotAnnot(ce, scope = msScope_qcSI, group = "TSO", normalise = FALSE, facet = "RT_PRIMERS")
plotAnnot(ce, scope = msScope_qcSI, group = "TSO", normalise = TRUE)
```

```{r}
plotAnnot(ce, scope = msScope_qcSI, group = "TSO_vol", normalise = FALSE)
```

```{r}
plotAnnot(ce, scope = msScope_qcSI, group = "PRIMERS_RATIO", normalise = FALSE, facet = "group")
```


Annotation with GENCODE
-----------------------

Collect Gencode annotations and gene symbols via a local GENCODE file
(mm9 gencode not available in AnnotationHub)

```{r annotate_CTSS}
annotateCTSS(ce, rtracklayer::import.gff("/osc-fs_home/scratch/gmtu/annotation/mus_musculus/gencode-M1/gencode.vM1.annotation.gtf.gz"))

plotAnnot(ce, scope = msScope_counts, group = "repl", facet = "group") +
  facet_wrap("facet", nrow = 5)
plotAnnot(ce, scope = msScope_counts, group = "repl", facet = "group", norm = F) +
  facet_wrap("facet", nrow = 5)
plotAnnot(ce, scope = msScope_counts, group = "TSO", normalise = FALSE, facet = "group")
```


CTSS ANALYSIS
=============

Number of nanoCAGE tags mapping at CTSS positions in each group of samples
--------------------------------------------------------------------------
  
```{r CTSS_counts_distribution}
plotReverseCumulatives(ce, onePlot = TRUE, fitInRange = NULL, legend = NULL)
```

Number of nanoCAGE tags mapping at CTSS positions in each sample
----------------------------------------------------------------
  
```{r ctss-analysis}
ce$l1 <- sapply(CTSStagCountDF(ce), function(x) sum(decode(x) > 0))
```

Gene expression analysis
========================

Make a gene expression table (not really required now).

```{r CTSStoGenes-CAGEexp}
CTSStoGenes(ce)
```

Promoter rate
=============

```{r promoter rate}
ce$promoter_rate <- ce$promoter / ce$librarySizes *100
```

Promoter rate
```{r plot promoter rate, dev=c('png'), fig.height=8, fig.width=10}
# dotsize <- 0.08
# ggplot(colData(ce) %>% data.frame, aes(x=RNA_group, y=promoter_rate)) +
#   stat_summary(fun.y=mean, fun.ymin=mean, fun.ymax=mean, geom="crossbar", color="gray") +
#   geom_dotplot(aes(fill=TSO), binaxis='y', binwidth=1.5, dotsize=dotsize, stackdir='center') + theme_bw() +
#   xlab("RNA group") +
#   ylab("Promoter rate") +
#   scale_x_continuous(breaks = c(1:70)) +
#   labs(title = "Promoter rate per RNA gropu") +
#   coord_flip()
```

Save myCAGEexp file.

```{r save myCAGEexp object}
#saveRDS(ce, "ce.Rds")
```

QC PLOTS
========

Boxplots
--------

### Extracted reads

```{r extracted_barcode, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(extracted ~ RNA_group, xlab = "RNA_group", ylab = "Extracted reads", data = colData(ce), cex.axis = 0.6, las = 3, main = "Number of extracted reads per sample")
```

```{r extracted_RT, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(extracted ~ RT_PRIMERS, ylab = "Extracted reads", xlab = "RT_PRIMERS concentration (uM)", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of extracted reads for different RT_PRIMERS concentrations")
```

```{r extracted_TSO, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(extracted ~ TSO, ylab = "Extracted reads", xlab = "TSO concentration (uM)", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of extracted reads for different TSO concentrations")
```

```{r extracted_ratio, dev=c('png')}
par(mar=c(7,5,2,2), cex.main = 1, font.main = 2)
boxplot(extracted ~ PRIMERS_RATIO, ylab = "Extracted reads", xlab = "Primers ratio", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of extracted reads for different primers ratio")
```

### Artefacts

```{r artefacts_barcode, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(tagdust / extracted ~ RNA_group, xlab = "RNA_group", ylab = "Artefacts/extracted reads", data = colData(ce), cex.axis = 0.6, las = 3, main = "Ratio artefacts/extracted per sample")
```

```{r artefacts_RT, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(tagdust / extracted ~ RT_PRIMERS, ylab = "Artefacts/extracted reads", xlab = "RT_PRIMERS concentration (uM)", data = colData(ce), cex.axis = 0.7, las = 3, main = "Ratio artefacts/extracted for different RT_PRIMERS concentrations")
```

```{r artefacts_TSO, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(tagdust / extracted ~ TSO, ylab = "Artefacts/extracted reads", xlab = "TSO concentration (uM)", data = colData(ce), cex.axis = 0.7, las = 3, main = "Ratio artefacts/extracted for different TSO concentrations")
```

```{r artefacts_ratio, dev=c('png')}
par(mar=c(7,5,2,2), cex.main = 1, font.main = 2)
boxplot(tagdust / extracted ~ PRIMERS_RATIO, ylab = "Artefacts/extracted reads", xlab = "Primers ratio", data = colData(ce), cex.axis = 0.7, las = 3, main = "Ratio artefacts/extracted for different primers ratio")
```

### Counts

```{r counts_barcode, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(librarySizes ~ RNA_group, xlab = "RNgroup", ylab = "Counts", data = colData(ce), cex.axis = 0.6, las = 3, main = "Number of read counts per sample")
```

```{r counts_RT, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(librarySizes ~ RT_PRIMERS, ylab = "Counts", xlab = "RT_PRIMERS concentration (uM)", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of read counts for different RT_PRIMERS concentrations")
```

```{r counts_TSO, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(librarySizes ~ TSO, ylab = "Counts", xlab = "TSO concentration (uM)", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of read counts for different TSO concentrations")
```

```{r counts_ratio, dev=c('png')}
par(mar=c(7,5,2,2), cex.main = 1, font.main = 2)
boxplot(librarySizes ~ PRIMERS_RATIO, ylab = "Counts", xlab = "Primers ratio", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of read counts for different primers ratio")
```

### Genes

```{r genes_barcode, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(genes ~ RNA_group, xlab = "RNA_group", ylab = "Genes", data = colData(ce), cex.axis = 0.6, las = 3, main = "Number of genes detected per sample")
```

```{r genes_RT, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(genes ~ RT_PRIMERS, ylab = "Genes", xlab = "RT_PRIMERS concentration (uM)", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of genes detected for different RT_PRIMERS concentrations")
```

```{r genes_TSO, dev=c('png')}
par(mar=c(5,5,2,2), cex.main = 1, font.main = 2)
boxplot(genes ~ TSO, ylab = "Genes", xlab = "TSO concentration (uM)", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of genes detected for different TSO concentrations")
```

```{r genes_ratio, dev=c('png')}
par(mar=c(7,5,2,2), cex.main = 1, font.main = 2)
boxplot(genes ~ PRIMERS_RATIO, ylab = "Genes", xlab = "Primers ratio", data = colData(ce), cex.axis = 0.7, las = 3, main = "Number of genes detected for different primers ratio")
```

Rarefaction
------------

```{r, message = FALSE}
rar1 <- hanabi(CTSStagCountDF(ce), from = 0)
#rarc <- hanabi(assay(consensusClustersSE(myCAGEexp)) %>% as.data.frame, from = 0)
rarg <- hanabi(assay(GeneExpSE(ce)), from = 0)
#save(rar1, rarg, file="rar.Rda") 
```

### Plot TSS discovery

```{r hanabi-TSS_byrep, dev=c('png')}
hanabiPlot(rar1, ylab='number of TSS detected', xlab='number of unique molecule counts', main=paste("TSS discovery"), group=ce$TSO, legend.pos = "bottomright")
```

### Plot Cluster discovery

```{r hanabi-CTSS_clusters_byrep, dev=c('png')}
#hanabiPlot(rarc, ylab='number of CTSS clusters detected', xlab='number of unique molecule counts', main=paste("Cluster discovery"), #group=myCAGEexp$group, legend.pos = "bottomright")
```

### Plot Gene discovery

```{r hanabi-gene_byrep, dev=c('png')}
hanabiPlot(rarg, ylab='number of genes detected', xlab='number of unique molecule counts', main=paste("Gene discovery"), group=ce$TSO, legend.pos = "bottomright")
```


```{r}
sessionInfo()
```