---
title: "Labcyte-RT Data Analysis (MOIRAI BED FILES)"
output:
  html_document:
    keep_md: yes
    number_sections: yes
    toc: yes
  pdf_document:
    toc: yes
---

```{r echo=FALSE}
knitr::opts_chunk$set(cache  = TRUE)
knitr::opts_knit$set(verbose = TRUE)
ggplot2::theme_set(ggplot2::theme_bw())
```

Data load and QC in R
=====================

```{r load_R_libraries}
library(CAGEr)
library(data.table)
library(dplyr)
library(ggplot2)
library(gplots)
library('RColorBrewer')
library(magrittr)
library(plyr)
library(MultiAssayExperiment)
library(SummarizedExperiment)
library(tibble)
library(reshape)
library(vegan)
data("nanoCAGE2017barcodes", package = "smallCAGEqc")
```

MOIRAI shortcuts

```{r moirai_shortcuts}
WORKFLOW       <- "OP-WORKFLOW-CAGEscan-short-reads-v2.1~rc1"
MOIRAI_PROJ    <- "project/Labcyte" 
MOIRAI_USER    <- "nanoCAGE2" 
ASSEMBLY       <- "mm9"
BASEDIR        <- "/osc-fs_home/scratch/moirai"
MOIRAI_BASE    <- file.path(BASEDIR, MOIRAI_USER)
MOIRAI_RESULTS <- file.path(MOIRAI_BASE, MOIRAI_PROJ)
```

Load CAGE libraries
===================

Load summary statistics from MOIRAI and polish the names
--------------------------------------------------------

```{r MOIRAI}
libs <- smallCAGEqc::loadMoiraiStats(
  pipeline  = "OP-WORKFLOW-CAGEscan-short-reads-v2.0",
  multiplex = file.path( MOIRAI_BASE, "input/171227_M00528_0321_000000000-B4GLP.multiplex.txt"),
  summary   = file.path( MOIRAI_RESULTS,"171227_M00528_0321_000000000-B4GLP.OP-WORKFLOW-CAGEscan-short-reads-v2.1~rc1.20180104125850/text/summary.txt"))
libs$barcode_ID <- libs$barcode
levels(libs$barcode_ID) %<>% match(nanoCAGE2017barcodes$barcodes)
libs$inputFiles <- list.files(path = "/osc-fs_home/scratch/moirai/nanoCAGE2/project/Labcyte/171227_M00528_0321_000000000-B4GLP.OP-WORKFLOW-CAGEscan-short-reads-v2.1~rc1.20180104125850/CAGEscan_fragments")
libs$inputFiles <- paste0("/osc-fs_home/scratch/moirai/nanoCAGE2/project/Labcyte/171227_M00528_0321_000000000-B4GLP.OP-WORKFLOW-CAGEscan-short-reads-v2.1~rc1.20180104125850/CAGEscan_fragments/", libs$inputFiles)
libs$inputFilesType <- c("bed")
libs$sampleLabels <- as.character(libs$samplename)
libs
```


```{r PLATE}
plate <- read.table("plate.txt", sep = "\t", header = TRUE)
plate_10ng <- subset(plate, plate$RNA == 10)
plate_10ng_no_RNA <- plate[224:230,]
plate_10ng_all <- rbind(plate_10ng, plate_10ng_no_RNA)
rm(plate) ; rm(plate_10ng) ; rm(plate_10ng_no_RNA)
plate_10ng_all
```

```{r LIBS}
libs <- cbind(libs, plate_10ng_all[match(libs$barcode, plate_10ng_all$BARCODE_SEQ),])
libs$PRIMERS_RATIO <- sub("no_RT_PRIMERS", "NA", libs$PRIMERS_RATIO)
libs$PRIMERS_RATIO <- as.numeric(libs$PRIMERS_RATIO)
libs$RNA <- as.numeric(libs$RNA)
libs <- libs[order(-libs$PRIMERS_RATIO),]
libs <- libs[order(-libs$RNA),]
libs$PRIMERS_RATIO[is.na(libs$PRIMERS_RATIO)] <- "no_RT_PRIMERS"
libs$plateID <- "A"
```

Control on barcode extraction
-----------------------------

```{r found_bc}
found_bc <- system("zcat /sequencedata/MiSeq/171227_M00528_0321_000000000-B4GLP/Data/Intensities/BaseCalls/1_S1_L001_R1_001.fastq.gz | sed -n 2~4p | cut -c1-6 | sort | uniq -c", intern = TRUE)
found_bc <- read.table(text=found_bc)
colnames(found_bc) <- c("n", "bc")
found_bc %<>% as.tibble %>% arrange(dplyr::desc(n))
```

```{r found_known_bcs}
libs$bc_extract_qc <- found_bc$n[match(libs$barcode, found_bc$bc)]

plot( libs$extracted
    , libs$bc_extract_qc
    , log = "xy"
    , xlab = "Extracted with TagDust 2"
    , ylab = "Counted directly from FASTQ file"
    , main = "QC of barcode extraction")
with(libs[libs$extracted / libs$bc_extract_qc > 2,], text(extracted * 2.5, bc_extract_qc, paste(barcode, barcode_ID)))
libs[libs$extracted / libs$bc_extract_qc > 2,]
```

```{r found_unknown_bcs}
found_bc$in_libs <- found_bc$bc %in% libs$barcode
found_bc$known   <- found_bc$bc %in% nanoCAGE2017barcodes$barcodes

hist(log10(found_bc$n), breaks = "fd")
rug(log10(libs$bc_extract_qc))
abline(v=log10(2e3), col = "grey")
```

True nanoCAGE barcodes that were not used in this experiment are very rare
in the sequence files.

```{r}
found_bc %>% filter(known, !in_libs) %>% as.data.frame()
```

Conversely, barcodes that are highly abundant but not used in the experiment
are not true nanoCAGE barcodes.  They may be related to higly abundant true
barcodes.

```{r}
found_bc %>% filter(!in_libs, n > 2e3) %>% as.data.frame

found_bc[found_bc$bc %in% nanoCAGE2017barcodes$barcodes[agrep("TGGGCA", nanoCAGE2017barcodes$barcodes, max = 2)],]

found_bc[found_bc$bc %in% nanoCAGE2017barcodes$barcodes[agrep("CCTGTC", nanoCAGE2017barcodes$barcodes, max = 1)],]

found_bc[found_bc$bc %in% nanoCAGE2017barcodes$barcodes[agrep("CACACA", nanoCAGE2017barcodes$barcodes, max = 1)],]
```

It might be possible that the HMM of TagDust has autocorrected some of the
errors, which would explain why some barcodes were detcted in larger amounts
with TagDust2.

Create a CAGEexp object and load expression data
------------------------------------------------

```{r new_CAGEexp, message = F, echo = TRUE}
ce <- CAGEexp(colData = DataFrame(libs)
                     , metadata = list( genomeName = "BSgenome.Mmusculus.UCSC.mm9"
                                        , inputFilesType = "bed"))

getCTSS(ce, useMulticore = TRUE)
removeStrandInvaders(ce)
normalizeTagCount(ce, method = "simpleTpm")
```

Save the CAGEexp file
---------------------

```{r save_myCAGEexp_object, dependson = "get_CTSS_data"}
saveRDS(ce, "Labcyte-RT_Data_Analysis.Rds")
```

Number of sequencing reads extracted per sample
-----------------------------------------------

```{r sample stats}
data.frame(colData(ce)[,"extracted",drop=F])
```


Quality controls
================

Custom _scopes_ displaying _strand invasion_ artefacts.

```{r custom_plotAnnot_scopes}
msScope_qcSI <- function(libs) {
  libs$Tag_dust     <- libs$extracted   - libs$rdna - libs$spikes - libs$cleaned
  libs$rDNA         <- libs$rdna
  libs$Spikes       <- libs$spikes
  libs$Unmapped     <- libs$cleaned     - libs$mapped
  libs$Non_proper   <- libs$mapped      - libs$properpairs
  libs$Duplicates   <- libs$properpairs - libs$librarySizes - libs$strandInvaders
  libs$Invaders     <- libs$strandInvaders
  libs$Counts       <- libs$librarySizes
  list( libs    = libs
      , columns = c( "Tag_dust", "rDNA", "Spikes", "Unmapped"
                   , "Non_proper", "Duplicates", "Invaders", "Counts")
      , total   = libs$extracted)
}

msScope_counts <- function(libs) {
  libs$Promoter   <- libs$promoter
  libs$Exon       <- libs$exon
  libs$Intron     <- libs$intron
  libs$Intergenic <- libs$librarySizes - libs$promoter - libs$intron - libs$exon
  libs$Invaders   <- libs$strandInvaders
  list( libs    = libs
      , columns = c("Promoter", "Exon", "Intron", "Intergenic", "Invaders")
      , total   = libs$librarySizes + libs$strandInvaders)
}
```

By RNA input
------------

```{r qc_RNA_quantity, dependson = "get_CTSS_data"}
ggpubr::ggarrange(
  plotAnnot( ce, scope = msScope_qcSI, group = "plateID", normalise = FALSE
           , title = "sequence counts"
           , facet="RNA") +
    facet_wrap(~facet, ncol = 1),
  plotAnnot( ce, scope = msScope_qcSI, group = "plateID", normalise = TRUE
         , title = "normalised to 100 %"
         , facet="RNA") +
  facet_wrap(~facet, ncol = 1),
  legend = "right",
  common.legend = TRUE
)
```


### Removal of the RNA negative controls

To ease data handling, the negative controls with no RNA are removed.

```{r remove_noRNA_ctls, dependson = "load_and_merge"}
ce <- ce[,ce$RNA != 0]
```


By RT primer quantity
---------------------

```{r qc_rt_primer_quantity, dependson = "remove_noRNA_ctls"}
ggpubr::ggarrange(
  plotAnnot( ce, scope = msScope_qcSI, group = "RT_PRIMERS", normalise = FALSE
           , title = "sequence counts"
           , facet="plateID") +
    facet_wrap(~facet, ncol = 1),
  plotAnnot( ce, scope = msScope_qcSI, group = "RT_PRIMERS", normalise = TRUE
         , title = "normalised to 100 %"
         , facet="plateID") +
  facet_wrap(~facet, ncol = 1),
  legend = "right",
  common.legend = TRUE
)
```


### Removal of the primer-negative controls

To ease data handling (for instance when working with primer ratios), the
negative controls with no primers are removed.

```{r remove_noPrimers_ctls, dependson = "remove_noRNA_ctls"}
ce <- ce[,ce$RT_PRIMERS != 0]
```

Lets tidy the primer ratios after the null primer concentrations were removed.

```{r correct_primer_ratios, dependson = "remove_noPrimers_ctls"}
ce$PRIMERS_RATIO %<>% factor
levels(ce$PRIMERS_RATIO) %<>% as.numeric %>% signif(2) %>% as.character
ce$PRIMERS_RATIO %<>% factor(levels = levels(ce$PRIMERS_RATIO) %>% gtools::mixedsort())
```

```{r qc_rt_primer_quantity_nozero, dependson = "remove_noRNA_ctls"}
ggpubr::ggarrange(
  plotAnnot( ce, scope = msScope_qcSI, group = "RT_PRIMERS", normalise = FALSE
           , title = "sequence counts"
           , facet="plateID") +
    facet_wrap(~facet, ncol = 1),
  plotAnnot( ce, scope = msScope_qcSI, group = "RT_PRIMERS", normalise = TRUE
         , title = "normalised to 100 %"
         , facet="plateID") +
  facet_wrap(~facet, ncol = 1),
  legend = "right",
  common.legend = TRUE
)
```


By TSO concentration
--------------------

```{r TSO, dependson = "get_CTSS_data"}
ggpubr::ggarrange(
  plotAnnot( ce, scope = msScope_qcSI, group = "TSO", normalise = FALSE
           , title = "sequence counts"
           , facet="plateID") +
    facet_wrap(~facet, ncol = 1),
  plotAnnot( ce, scope = msScope_qcSI, group = "TSO", normalise = TRUE
         , title = "normalised to 100 %"
         , facet="plateID") +
  facet_wrap(~facet, ncol = 1),
  legend = "right",
  common.legend = TRUE
)
```


Create CTSS clusters
--------------------

```{r create_CTSS_clusters, message = FALSE, warning=FALSE}
# clusterCTSS(ce, thresholdIsTpm = FALSE, useMulticore = TRUE, nrPassThreshold = 2, removeSingletons = TRUE)
# cumulativeCTSSdistribution(ce, clusters = "tagClusters")
# #, use multicore = TRUE)
# quantilePositions(ce, clusters = "tagClusters", qLow = 0.1, qUp = 0.9, useMulticore = TRUE)
```


Annotation with GENCODE
-----------------------

```{r annotate_CTSS, dependson = "get_CTSS_data"}
annotateCTSS(ce, rtracklayer::import.gff("/osc-fs_home/scratch/gmtu/annotation/mus_musculus/gencode-M1/gencode.vM1.annotation.gtf.gz"))
```


```{r CTSS_annotation_plot_per_replicate, dependson = "annotate_CTSS"}
ggpubr::ggarrange(
  plotAnnot( ce, scope = msScope_counts, group = "TSO", normalise = FALSE
           , title = "sequence counts"
           , facet="RT_PRIMERS") +
    facet_wrap(~facet, ncol = 1),
  plotAnnot( ce, scope = msScope_counts, group = "TSO", normalise = TRUE
         , title = "normalised to 100 %"
         , facet="RT_PRIMERS") +
  facet_wrap(~facet, ncol = 1),
  legend = "right",
  common.legend = TRUE
)
```

Make a gene expression table (not really required now).

```{r CTSStoGenes-CAGEexp}
CTSStoGenes(ce)
#CTSScoordinatesGR(ce)
```
