---
title: "Comparison of SuperScript III and IV for different concentrations of TSO and RTP"
output: 
  html_document: 
    keep_md: yes
    number_sections: yes
    toc: yes
editor_options: 
  chunk_output_type: console
---

```{r echo=FALSE}
knitr::opts_chunk$set(cache  = TRUE)
knitr::opts_knit$set(verbose = TRUE)
ggplot2::theme_set(ggplot2::theme_bw())
```

Data load and QC in R
=====================

```{r load_R_libraries, message=F}
library(AnnotationHub)
BS_GENOME    <- "BSgenome.Hsapiens.plusHPV.hg38" 
library(BS_GENOME, character.only = T)
library(CAGEr)
library(data.table)
library(ggplot2)
library(magrittr)
library(plyr)
library(MultiAssayExperiment)
library(SummarizedExperiment)
library(reshape)
library(vegan)
source("loadMoiraiStats.R")
```


Load CAGE libraries
===================

Load summary statistics from MOIRAI and polish the names.
---------------------------------------------------------

### NC 33

```{r NC33_MOIRAI}
libs <- loadMoiraiStats(
  pipeline  = "OP-WORKFLOW-CAGEscan-short-reads-v2.0",
  multiplex = file.path( "multiplex_files/NC33.multiplex.txt"),
  summary   = file.path( "NC33.OP-WORKFLOW-CAGEscan-short-reads-v2.0.20170119161243/text/summary.txt"))

libs$group  <- sub("SSC3",     "SSIII", libs$group)
libs$group  <- sub("cont3", "NC SSIII", libs$group)
libs$group  <- sub("SSC4",      "SSIV", libs$group)
libs$group  <- sub("cont4",  "NC SSIV", libs$group)
libs$group2 <- libs$group
libs$group  <- paste(libs$group, "exp1")

MOIRAI_FRAGS <-
 file.path( "NC33.OP-WORKFLOW-CAGEscan-short-reads-v2.0.20170119161243/CAGEscan_fragments") %>%
    list.files(full.names = TRUE)

names(MOIRAI_FRAGS) <- sub(".bed", "", basename(MOIRAI_FRAGS))

libs <- libs[names(MOIRAI_FRAGS), ]
libs$sampleLabels   <- names(MOIRAI_FRAGS)
libs$inputFiles     <- MOIRAI_FRAGS
libs$inputFilesType <- "bed"

rm(MOIRAI_FRAGS)

libs_all <- libs
```

### NC37

```{r NC37_MOIRAI}
libs <- smallCAGEqc::loadMoiraiStats(
  pipeline  = "OP-WORKFLOW-CAGEscan-short-reads-v2.0",
  multiplex = file.path( "multiplex_files/NC37.multiplex.txt"),
  summary   = file.path( "NC37.OP-WORKFLOW-CAGEscan-short-reads-v2.0.20170119161306/text/summary.txt"))

libs$group  <- sub("SSC3",     "SSIII", libs$group)
libs$group  <- sub("cont3", "NC SSIII", libs$group)
libs$group  <- sub("SSC4",      "SSIV", libs$group)
libs$group  <- sub("cont4",  "NC SSIV", libs$group)
libs$group2 <- libs$group
libs$group  <- paste(libs$group, "exp2")

MOIRAI_FRAGS <-
 file.path( "NC37.OP-WORKFLOW-CAGEscan-short-reads-v2.0.20170119161306/CAGEscan_fragments") %>%
    list.files(full.names = TRUE)

names(MOIRAI_FRAGS) <- sub(".bed", "", basename(MOIRAI_FRAGS))

libs <- libs[names(MOIRAI_FRAGS), ]
libs$sampleLabels   <- names(MOIRAI_FRAGS)
libs$inputFiles     <- MOIRAI_FRAGS
libs$inputFilesType <- "bed"

rm(MOIRAI_FRAGS)

libs_all <- rbind(libs_all, libs)
```

### NC38

```{r NC38_MOIRAI}
libs <- smallCAGEqc::loadMoiraiStats(
  pipeline  = "OP-WORKFLOW-CAGEscan-short-reads-v2.0",
  multiplex = file.path( "multiplex_files/NC38.multiplex.txt"),
  summary   = file.path( "NC38.OP-WORKFLOW-CAGEscan-short-reads-v2.0.20170119161330/text/summary.txt"))

libs$group <- sub("non_fixed", "SSIV high TSO", libs$group)
libs$group <- sub("cont_NF", "NC SSIV high TSO", libs$group)
libs$group2 <- libs$group
libs$group  <- paste(libs$group, "exp3")

MOIRAI_FRAGS <-
 file.path( "NC38.OP-WORKFLOW-CAGEscan-short-reads-v2.0.20170119161330/CAGEscan_fragments/") %>%
    list.files(full.names = TRUE)

names(MOIRAI_FRAGS) <- sub(".bed", "", basename(MOIRAI_FRAGS))

libs <- libs[names(MOIRAI_FRAGS), ]
libs$sampleLabels   <- names(MOIRAI_FRAGS)
libs$inputFiles     <- MOIRAI_FRAGS
libs$inputFilesType <- "bed"

rm(MOIRAI_FRAGS)

libs_all <- rbind(libs_all, libs)
```

### NC39

```{r NC39_MOIRAI}
libs <- smallCAGEqc::loadMoiraiStats(
  pipeline  = "OP-WORKFLOW-CAGEscan-short-reads-v2.0",
  multiplex = file.path( "multiplex_files/NC39.multiplex.txt"),
  summary   = file.path( "NC39.OP-WORKFLOW-CAGEscan-short-reads-v2.0.20170119161348/text/summary.txt"))

libs$group <- sub("SSC4", "SSIV", libs$group)
libs$group <- sub("cont", "NC SSIV", libs$group)
libs$group2 <- libs$group
libs$group  <- paste(libs$group, "exp4?")

MOIRAI_FRAGS <-
 file.path( "NC39.OP-WORKFLOW-CAGEscan-short-reads-v2.0.20170119161348/CAGEscan_fragments/") %>%
    list.files(full.names = TRUE)

names(MOIRAI_FRAGS) <- sub(".bed", "", basename(MOIRAI_FRAGS))

libs <- libs[names(MOIRAI_FRAGS), ]
libs$sampleLabels   <- names(MOIRAI_FRAGS)
libs$inputFiles     <- MOIRAI_FRAGS
libs$inputFilesType <- "bed"

rm(MOIRAI_FRAGS)

libs_all <- rbind(libs_all, libs)
libs <- libs_all
```


### Remove empty files

```{r removeEmptyFiles}
libs <- libs[file.info(libs$inputFiles)$size > 0,]
```

### Create a CAGEexp object and load expression data.

```{r new_CAGEexp, message = FALSE}
myCAGEexp <- new( "CAGEexp"
                , colData = DataFrame(libs)
                , metadata = list( genomeName = BS_GENOME
                                 , inputFilesType = "bed"))
rm(libs)
getCTSS(myCAGEexp)
```

CTSS analysis
=============

```{r ctss-analysis}
myCAGEexp$l1     <- colSums(CTSStagCountDf(myCAGEexp) > 0)
```

Annotation with GENCODE
-----------------------

```{r get-annotation}
# ah <- AnnotationHub()
# query(ah, c("Gencode", "gff", "human"))
# gff <- ah[["AH49556"]]
# EBI's FTP server borken today, let's use ENSEMBL instead
# ftp://ftp.ensembl.org/pub/release-97/gtf/homo_sapiens/Homo_sapiens.GRCh38.97.gtf.gz
gff <- rtracklayer::import.gff("Homo_sapiens.GRCh38.97.gtf.gz")
seqlevels(gff) <- paste0("chr", seqlevels(gff))
```

Annotate the genomic ranges of the `tagCountMatrix` SummarizedExperiment.

```{r annotate-CAGEexp}
annotateCTSS(myCAGEexp, gff)
CTSStoGenes(myCAGEexp)
```


Create of a SummarizedExperiemnt for gene expression level.

```{r}
myCAGEexp$genes <- colSums(assay(GeneExpSE(myCAGEexp)) %>% as.data.frame > 0)
#myCAGEexp$geneSymbols <- countSymbols(assay(GeneExpSE(myCAGEexp)) %>% as.data.frame)
```

Richness
--------

```{r calculate-richness}
myCAGEexp$r100l1 <- rarefy(t(CTSStagCountDf(myCAGEexp) %>% as.matrix),100)
myCAGEexp$r100l1[myCAGEexp$counts < 100] <- NA
```


Analysis
=========

### Removing cells with less than 100 counts

```{r}
myCAGEexp$QCed <- myCAGEexp$counts >= 100
```

### Number of reads/single cells

```{r}
with(colData(myCAGEexp)[myCAGEexp$QCed,],
  data.frame( mean = tapply(extracted, group, mean)
            ,   sd = tapply(extracted, group, sd)
            ,    n = tapply(extracted, group, length)))
```

Produce libs table without negative controls to draw cleaner hanabiPlot.

Also removed fixed cells as they are not relevant in this analysis.

```{r}
myCAGEexp$keep <- (! grepl("^NC", myCAGEexp$group)) & (! grepl("fixed", myCAGEexp$group)) & myCAGEexp$QCed
CE <- myCAGEexp[, myCAGEexp$keep]
CD <- droplevels(colData(CE))
colData(CE) <- CD
```


### Graph

#### Gene count

```{r gene-count2, dev=c('png', 'svg'), fig.height=2.5}
dotsize <- 30
ggplot(CD %>% data.frame, aes(x=group2, y=genes)) +
  stat_summary(fun.y=mean, fun.ymin=mean, fun.ymax=mean, geom="crossbar", color="gray") +
  geom_dotplot(aes(fill=group), binaxis='y', binwidth=1, dotsize=dotsize, stackdir='center') + theme_bw() +
  xlab("Enzyme") +
  coord_flip()
```

```{r gene-count, dev=c('png', 'svg'), fig.height=2.5}
dotsize <- 30
ggplot(CD %>% data.frame, aes(x=group, y=genes)) +
  stat_summary(fun.y=mean, fun.ymin=mean, fun.ymax=mean, geom="crossbar", color="gray") +
  geom_dotplot(aes(fill=group), binaxis='y', binwidth=1, dotsize=dotsize, stackdir='center') + theme_bw() +
  coord_flip()
```

#### Transcripts counts

```{r transcripts-count2, dev=c('png', 'svg'), fig.height=2.5}
dotsize <- 120
ggplot(CD %>% data.frame, aes(x=group2, y=counts)) +
  stat_summary(fun.y=mean, fun.ymin=mean, fun.ymax=mean, geom="crossbar", color="gray") +
  geom_dotplot(aes(fill=group), binaxis='y', binwidth=1, dotsize=dotsize, stackdir='center') + theme_bw() +
  xlab("Enzyme") +
  coord_flip()
```

```{r transcripts-count, dev=c('png', 'svg'), fig.height=2.5}
dotsize <- 120
ggplot(CD %>% data.frame, aes(x=group, y=counts)) +
  stat_summary(fun.y=mean, fun.ymin=mean, fun.ymax=mean, geom="crossbar", color="gray") +
  geom_dotplot(aes(fill=group), binaxis='y', binwidth=1, dotsize=dotsize, stackdir='center') + theme_bw() +
  xlab("Enzyme") +
  coord_flip()
```

### PlotAnnot

```{r plotAnnot, dev=c('png', 'svg'), fig.height=2.5}
plotAnnot(CE, scope="counts", group="group")
plotAnnot(CE, scope="counts", group="group2")
plotAnnot(CE, scope="qc", group="group") + scale_fill_brewer(palette = "Set2")
plotAnnot(CE, scope="qc", group="group2")+ scale_fill_brewer(palette = "Set2")
```

### Rarefaction

Raw reads per molecule (BED12 data)
-----------------------------------

```{r load_fragments}
bed <- sapply(inputFiles(myCAGEexp), rtracklayer::import.bed)
names(bed) <- sampleLabels(myCAGEexp)
bed <- bed[myCAGEexp$keep]
```

```{r, message=FALSE}
library(vegan)
# If applicable, restore the previous computation with the command below:
# load("rar.Rda")

# Rarefy at with enough sampling points to give a smooth appearance to
# the curves.  It can take time !

rar1 <- hanabi(CTSStagCountDF(CE), from = 0)
rarg <- hanabi(assay(GeneExpSE(CE)) %>% as.data.frame, from = 0)
rarU <- hanabi(bed, from = 0)

# Save the result on the hard drive.  This way, if the commands have
#been run
# through knitr, the long computations here can be skipped if needed
#again in
# an interactive session.
save(rar1, rarg, rarU, file="rar.Rda") 
```

### Graphs

#### Plot TSS discovery

```{r hanabi-TSS, dev=c('png', 'svg')}
hanabiPlot(rar1, ylab='number of TSS detected', xlab='number of unique molecule counts', main=paste("TSS discovery"), group=CD$group2)
```

```{r hanabi-TSS_byrep, dev=c('png', 'svg')}
hanabiPlot(rar1, ylab='number of TSS detected', xlab='number of unique molecule counts', main=paste("TSS discovery"), group=CD$group)
```

#### Plot Gene discovery

```{r hanabi-gene, dev=c('png', 'svg')}
hanabiPlot(rarg, ylab='number of genes detected', xlab='number of unique molecule counts', main=paste("Gene discovery"), group=CD$group2)
```

```{r hanabi-gene_byrep, dev=c('png', 'svg')}
hanabiPlot(rarg, ylab='number of genes detected', xlab='number of unique molecule counts', main=paste("Gene discovery"), group=CD$group)
```

#### Plot Molecule discovery (with raw reads)

```{r hanabi-UMI, dev=c('png', 'svg')}
hanabiPlot(rarU, ylab='number of molecules detected', xlab='number of properly mapped reads', main=paste("Transcript discovery "), group=CD$group2)
```

```{r hanabi-UMI2, dev=c('png', 'svg')}
hanabiPlot(rarU, ylab='number of molecules detected', xlab='number of properly mapped reads', main=paste("Transcript discovery"), group=CD$group, legend.pos = "topright")
```

#### Richness

```{r richness, dev=c('png', 'svg')}
qplot(data = CD %>% as.data.frame, group, r100l1, geom = c("boxplot", "jitter"))
qplot(data = CD %>% as.data.frame, group2, r100l1, geom = c("boxplot", "jitter"))
```